code/game/machinery/doors/airlock_control.dm

```diff
@@ -165,7 +165,9 @@
 		var/datum/gas_mixture/air_sample = return_air()
 		var/new_pressure = round(air_sample.return_pressure(),0.1)
 
-		if(abs(pressure - new_pressure) > 0.001 || pressure == null)
+		//Try preventing the sensor from spamming when there's no differences, and the controller isn't cycling.
+		//Can't rely only on difference, since it's possible for the pressure to stay nearly the same for a few calls, which prevents the machine var from updating
+		if(abs(pressure - new_pressure) > 0.001 || master_cycling)
```

Comment:
Are you sure that this is needed in light of the other change? `master_cycling` is an icon update var which is logically decoupled from pressure updating. Because the public vars/methods are in principle user-accessible, it's desirable to avoid these kinds of opaque internal dependencies.

I think adding `|| new_pressure == null` would be fine and am surprised it doesn't do that, but is probably inconsequential (will report next tick anyway).

Anything beyond that, I would prefer to modify the logic on the airlock controller instead.

Reply:
Yes. master_cycling was never meant to be only for icon updates. I added that to prevent all the airlock machines from going out of sync which they used to do constantly before for a variety of reasons. 

While the master is cycling, you want sensors to prioritize sending pressure updates, even if it's below the threshold. Because otherwise the master controller will never know it actually emptied the chamber, unless it gets really lucky and catch the last pressure update when the chamber is below the threshold. That's the bug this has been fixing downstream for a while.

Also, ``pressure == null`` must be communicated as well, because otherwise the the controller will be sent only the last value before the pressure actually hit 0/null, which could be anything depending on how fast the pumps remove the air from the chamber. We have airlocks with a bunch of high capacity vents downstream and this is what was happening, it was pulling air out too fast for the sensors to update before pressure hit 0, and the controller would get stuck with the last chamber pressure before it hit 0, never completing it's purge.

Reply:
But it _does_ send when `pressure == null` already. I don't follow. Of course, when it sends it, it's not received, because of the bug in the other line you touch, but it is clearly sent.

Reply:
0 != null, so that condition already doesn't work. Null doesn't even gets cast to 0 in that check.
So it would not update when we hit 0, until we hit null, which isn't a given and only happens when there's no moles left of anything, iirc.

But it's true that master_cycling might not be the best replacement either. The condition here was mainly a way to try to lower the cost of having a bunch of airlock sensors sending tons of pressure updates constantly. But it doesn't really do a good job at it.

Reply:
If it's set to `0` instead of a null literal, then check `pressure` instead of `pressure == null`.

Reply:
I'll just remove that if entirely.